File: Sort/src/sort/SelectionSort.java
package sort;

/**
 * Selection Sort (选择排序)
 * 
 * @author fantasy
 */
public class SelectionSort {

    public static void selectionSort(int[] data) {

        for (int i = 0; i < data.length; i++) {
            int largest = i;
            int temp;
            for (int j = i + 1; j < data.length; j++) {
                if (data[j] > data[largest]) {
                    largest = j;
                }
            }
            temp = data[i];
            data[i] = data[largest];
            data[largest] = temp;
        }
    }
}

File: SortJS/SelectionSort.js
/**
 *
 * SelectionSort
 *
 * @description
 * @author Fantasy <fantasyshao@icloud.com>
 * @create 2014-10-11
 * @update 2014-10-11
 */

module.exports = function (arr) {

  console.log('\nSelection sort...');

  var i, j, min;

  for (i = 0; i < arr.dataStore.length; i++) {
    min = i;

    for (j = i + 1; j < arr.dataStore.length; j++) {
      if (arr.dataStore[min] > arr.dataStore[j]) {
        min = j;
      }
    }

    if (min !== i) {
      arr.swap(arr.dataStore, min, i);
    }
  }

  return arr.dataStore;

};